components/provider_cldr/src/transform/dates.rs

```diff
@@ -353,6 +355,8 @@ pub(self) mod cldr_json {
         day_periods,
         ["am", am, Cow<'static, str>],
         ["pm", pm, Cow<'static, str>],
+        ["noon", noon, Option<Cow<'static, str>>],
+        ["midnight", midnight, Option<Cow<'static, str>>],
```

Comment:
can you use the same trick we use in Plurals to omit serialization of optional fields here: https://github.com/unicode-org/icu4x/blob/master/components/provider/src/structs/plurals.rs#L36-L37

This will make the resources smaller.

Reply:
In fact, it would be really cool to skip widths that fallback onto other widths that are exactly the same. For example for French it seems like we could have just one width stored since all the specializations are the same.
@sffc - wdyt?

(not a blocker, may be worth filing a separate issue to do the same for DateTime)

Reply:
> In fact, it would be really cool to skip widths that fallback onto other widths that are exactly the same. For example for French it seems like we could have just one width stored since all the specializations are the same.

This is in scope of #257.

Reply:
Sounds reasonable to do this in a separate PR, then.